sympy/series/series.py

```diff
@@ -61,3 +62,110 @@ def series(expr, x=None, x0=0, n=6, dir="+"):
+
+    .. math::
+        g(y) = a + \sum_{n = 1}^{oo} \frac{g_n (y - f(a))^n}{n!}
+
+    where
+
+    .. math::
+        g_n = \lim_{y \to a} \left[ \frac{d^{n-1}}{dx^{n-1}}
+        \left( \frac{x-a}{f(x)-f(a)} \right) ^n \right]
+
+    The theorem further states that this series has a non-zero radius of convergence,
+    i.e., x represents an analytic function of y in a neighbourhood of y = f(a).
+
+    Parameters
+    ==========
+
+    f : expression, the expansion of whose inverse is to be found.
+
+    x : variable, in terms of which equation is given, and in terms of which, the
+    expansion will be provided.
+
+    a : point on x-axis in the neighbourhood of which, the inverse expansion is to be found.
+
+    n : no of terms required in the expansion; default value is 3 terms.
+
+    Examples
+    ========
+
+    >>> from sympy.series.series import inversion_series
+    >>> from sympy import exp
+    >>> from sympy.abc import x
+
+    Let the equation whose inverse's expansion is to be found be y = f(x)
+    then the first argument to be supplied must be f(x).
+
+    >>> f = exp(x)
+
+    In this case, the inverse series is that of the logarithmic function around 1 = f(0):
+
+    >>> inversion_series(f,x,0,4)
+    x + (x - 1)**3/3 - (x - 1)**2/2 - 1
+
+    Functions with vanishing derivative cannot be inverted with this method
+    >>> f = x**2
+    >>> try:
+    ...    inversion_series(f,x,0,2)
+    ... except ValueError as ex:
+    ...     print(repr(ex))
+    ValueError('The inversion formula requires non-zero derivative.')
+
+    Note
+    ====
+
+    There is a straightforward derivation using complex analysis and contour integration;
+    the complex formal power series version is a consequence of knowing the formula for polynomials,
+    so the theory of analytic functions may be applied. Actually, the machinery from analytic
+    function theory enters only in a formal way in this proof, in that what is really needed is
+    some property of the formal residue, and a more direct formal proof is available.
+
+    The implementation uses a quasi-newton method to solve an implicit equation.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Lagrange_inversion_theorem
+    .. [2] http://mathworld.wolfram.com/LagrangeInversionTheorem.html
+    .. [3] http://www.cfm.brown.edu/people/dobrush/am33/Mathematica/lit.html
+       [4] Abramowitz, M. and Stegun, I.A., 1964. "Handbook of Mathematical Functions", Eq. 3.6.6
+    """
+
+    if f.diff(x,1).subs({x:a}) == 0:
+        raise ValueError("The inversion formula requires non-zero derivative.")
+
+    # solve using quasi-newton method
+    fa = f.subs({x:a})
+    f = f.series(x,a,n).removeO()
+    dfa = f.diff(x,1).subs({x:a})
+
+    y = symbols('y')
+    x_series = a + (y-fa)/dfa
+
+    for i in range(2,n):
+        step = f.subs({x:x_series}).series(y,fa,i+1).coeff(y-fa,i)
+        x_series = x_series-step*(y-fa)**i/dfa
```

Comment:
This seems to be very inefficient:
```python
In [4]: %time inversion_series(exp(x), x, 0, 10)
CPU times: user 10.2 s, sys: 8 ms, total: 10.2 s
Wall time: 10.2 s
Out[4]: 
           9          8          7          6          5          4          3          2    
    (x - 1)    (x - 1)    (x - 1)    (x - 1)    (x - 1)    (x - 1)    (x - 1)    (x - 1)     
x + ──────── - ──────── + ──────── - ──────── + ──────── - ──────── + ──────── - ──────── - 1
       9          8          7          6          5          4          3          2 
```

Reply:
Apple vs oranges. 
The current proposal work with expressions (EX domain?), rather than in a poly ring.

Try to solve this exercise, or something more advanced
```
# Chernoff bounds for beta disribution

import sympy as sm
from sympy.series.series import inversion_series

c,v,t,eps = sm.symbols('c,v,t,epsilon')
MGF = 1+v*t**2/2 + v*c*t**3/6
crammer = t*eps-sm.log(MGF)
eps_t = sm.solve(crammer.diff(t), eps)[0]
t_eps = inversion_series(eps_t,t,0,3).subs({t:eps})
t_eps # - \frac{c \epsilon^{2}}{2 v^{2}} + \frac{\epsilon}{v}
```

Reply:
There is also EXRAW which can be used here.

Reply:
The speed is fixed now.